File: backend/mpp1/Model/VehicleRent.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace mpp1.Model;

[Table("VehicleRent")]
public class VehicleRent : IValidatableObject
{
    [Key] public Guid Id { get; set; }
    public virtual Vehicle? Vehicle { get; set; }
    public virtual Client? Client { get; set; }
    public virtual Guid VehicleId { get; set; }
    public virtual Guid ClientId { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }

    [JsonIgnore]
    [InverseProperty("VehicleRents")]
    public virtual User? User { get; set; } = null!;

    [ForeignKey("User")] public Guid UserId { get; set; }

    public int TotalCost { get; set; }
    public string? Comments { get; set; }

    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        if (StartDate > EndDate)
        {
            yield return new ValidationResult("Start date can not be later then end date!", new[] { "errors   " });
        }

        if (TotalCost < 0)
        {
            yield return new ValidationResult("Total cost can not be lower than 0!", new[] { "errors" });
        }
    }
}

File: backend/mpp1/dbscripts/model/VehicleRent.py
from Utilities import get_string_in_single_quotes
class VehicleRent():
    def __init__(self, id, client_id, vehicle_id, start_date, end_date, total_cost, comments, user_id):
        self.__id = id
        self.__client_id = client_id
        self.__vehicle_id = vehicle_id
        self.__start_date = start_date
        self.__end_date = end_date
        self.__total_cost = total_cost
        self.__comments = comments
        self.__user_id = user_id

    def __str__(self) -> str:
        return f"({get_string_in_single_quotes(self.__id)}," + \
               f"{get_string_in_single_quotes(self.__client_id)}," + \
               f"{get_string_in_single_quotes(self.__vehicle_id)}," + \
               f"{get_string_in_single_quotes(self.__start_date)}," + \
               f"{get_string_in_single_quotes(self.__end_date)}," + \
               f"{self.__total_cost}," + \
               f"{get_string_in_single_quotes(self.__comments)}," + \
               f"{get_string_in_single_quotes(self.__user_id)})"
               